edb/schema/name.py

```diff
@@ -62,14 +73,16 @@ def as_tuple(self):
         return (self.module, self.name)
 
     @staticmethod
-    def is_qualified(name):
+    def is_qualified(name: Union[SchemaName, str]) -> bool:
         return isinstance(name, SchemaName) or '::' in name
 
 
 class UnqualifiedName(SchemaName):
 
-    def __new__(cls, name):
-        result = str.__new__(cls, name)
+    def __new__(cls: Type[NameT], name: str) -> NameT:
+        # Ignore below since Mypy doesn't believe you can pass `name` to
+        # object.__new__.
```

Comment:
This is a bug of mypy, right?

Reply:
Yes and no. `object` itself does not accept kwargs since Python 2.0 or something:
```pycon
>>> object.__new__(object, name='hey')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object() takes no arguments
```
But in this case `super()` will actually not be `object` yet because we're using multiple inheritance and Mypy doesn't see it.